backend/server.js

```diff
@@ -1,24 +1,128 @@
+import bcrypt from "bcrypt";
 import cors from "cors";
+import dotenv from "dotenv";
 import express from "express";
+import expressListEndpoints from "express-list-endpoints";
 import mongoose from "mongoose";
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
+dotenv.config();
+const { Schema } = mongoose;
+const mongoUrl =
+  process.env.MONGO_URL || "mongodb://localhost/project-authentication";
 mongoose.connect(mongoUrl);
 mongoose.Promise = Promise;
 
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
+const userSchema = new Schema({
+  username: { type: String, unique: true, required: true, minLength: 8 },
+  email: {
+    type: String,
+    unique: true,
+    required: true,
+    match: /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/,
+  },
+  password: { type: String, required: true },
+  accessToken: { type: String, default: () => bcrypt.genSaltSync() },
+});
+
+const User = mongoose.model("User", userSchema);
+
 const port = process.env.PORT || 8080;
 const app = express();
 
+// Function for authenticate by token
+const authenticateUser = async (req, res, next) => {
+  const user = await User.findOne({ accessToken: req.header("Authorization") });
+  if (user) {
+    req.user = user;
+    next();
+  } else {
+    res.status(401).json({
+      message: "Unauthorised access",
+      success: false,
+      error: "You are not allowed to see our top secret message!",
+    });
+  }
+};
+
 // Add middlewares to enable cors and json body parsing
-app.use(cors());
+app.use(
+  cors({
+    origin: "https://team-peace-auth.netlify.app",
+    methods: ["GET", "POST"],
+  }) // Allow sending credentials from frontend to backend
+);
 app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  const endpoints = expressListEndpoints(app);
+  res.json(endpoints);
+});
+
+// Sign-up
+app.post("/signup", async (req, res) => {
+  try {
+    const { username, email, password } = req.body;
+    const user = new User({
+      username,
+      email,
+      password: bcrypt.hashSync(password, 10),
+    });
+    await user.save();
+    res.status(201).json({ message: "Sign up successfully", success: true });
+  } catch (error) {
+    console.log(error.message);
+    res.status(400).json({
+      message: "Could not sign up.",
+      success: false,
+      error: error.message,
+    });
+  }
+});
+
+// Log-in
+app.post("/login", async (req, res) => {
```

Comment:
Usually endpoints are named in plural nouns, after what they return or in this case create. So a name suggestion is: POST /users and POST /sessions